src/vivarium_gates_iv_iron/components/hemoglobin.py

```diff
@@ -163,8 +164,8 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
         self.population_view.update(pop_update)
 
     def hemoglobin_source(self, idx: pd.Index) -> pd.Series:
-        distribution_parameters = self.distribution_parameters(idx)
         pop = self.population_view.get(idx)
+        distribution_parameters = self.distribution_parameters(pop.index)
```

Comment:
distribution_parameters would originally pull data for untracked simulants unlike the population view. This change ensures that the state table and distribution parameters share the same index.